app/src/main/java/swati4star/createpdf/util/RealPathUtil.java

```diff
@@ -30,53 +32,122 @@ public static String getRealPath(Context context, Uri fileUri) {
             } else if (isDownloadsDocument(uri)) {
+                path = getDownloadsDocumentPath(context, uri, false);
+            }
+        }
+        return path;
+    }
 
-                final String id = DocumentsContract.getDocumentId(uri);
+    /**
+     * Get a file path from an Uri that points to the Downloads folder.
+     *
+     * @param context       The context
+     * @param uri           The uri to query
+     * @param hasSubFolders The flag that indicates if the file is in the root or in a subfolder
+     * @return The absolute file path
+     */
+    private static String getDownloadsDocumentPath(Context context, Uri uri, boolean hasSubFolders) {
+        String fileName = getFilePath(context, uri);
+        String subFolderName = hasSubFolders ? getSubFolders(uri) : "";
 
-                if (!TextUtils.isEmpty(id)) {
-                    if (id.startsWith("raw:")) {
-                        path =  id.replaceFirst("raw:", "");
-                    }
-                    try {
-                        final Uri contentUri = ContentUris.withAppendedId(
-                                Uri.parse("content://downloads/public_downloads"), Long.valueOf(id));
+        if (fileName != null) {
+            return Environment.getExternalStorageDirectory().toString() +
+                    "/Download/" + subFolderName + fileName;
+        }
+        final String id = DocumentsContract.getDocumentId(uri);
 
-                        path = getDataColumn(context, contentUri, null, null);
-                    } catch (NumberFormatException e) {
-                        e.printStackTrace();
-                    }
-                }
+        String path = null;
+        if (!TextUtils.isEmpty(id)) {
+            if (id.startsWith("raw:")) {
+                path = id.replaceFirst("raw:", "");
+            }
+            try {
+                final Uri contentUri = ContentUris.withAppendedId(
+                        Uri.parse("content://downloads/public_downloads"), Long.valueOf(id));
+                path = getDataColumn(context, contentUri, null, null);
+            } catch (NumberFormatException e) {
+                e.printStackTrace();
             }
+        }
+        return path;
+
+    }
+
+    /**
+     * Get all the subfolders from an Uri.
+     *
+     * @param uri The uri
+     * @return A string containing all the subfolders that point to the final file path
+     */
+    private static String getSubFolders(Uri uri) {
+        String replaceChars = String.valueOf(uri).replace("%2F", "/")
+                .replace("%20", " ").replace("%3A", ":");
+        String[] bits = replaceChars.split("/");
+        String sub5 = bits[bits.length - 2];
+        String sub4 = bits[bits.length - 3];
+        String sub3 = bits[bits.length - 4];
+        String sub2 = bits[bits.length - 5];
+        String sub1 = bits[bits.length - 6];
+        if (sub1.equals("Download")) {
```

Comment:
It isnt exactly clear what are you trying to do here?
Add comments.

Reply:
is it clear now? although I have to test it with longer paths and subfolders.